src/fparser/two/Fortran2003.py

```diff
@@ -3958,16 +3958,47 @@ def match(string):
         return SeparatorBase.match(Scalar_Int_Expr, Scalar_Int_Expr, string)
 
 
-class Data_Ref(SequenceBase):  # R612
-    """
-    <data-ref> = <part-ref> [ % <part-ref> ]...
-    """
+class Data_Ref(SequenceBase):
+    '''
+    Fortran 2003 Rule R612
+
+    data-ref is part-ref [ % part-ref ] ...
+
+    '''
     subclass_names = ['Part_Ref']
     use_names = []
 
+    @staticmethod
     def match(string):
-        return SequenceBase.match(r'%', Part_Ref, string)
-    match = staticmethod(match)
+        '''Implements the matching for a data-reference. This defines a series
+        of dereferences e.g. a%b%c.
+
+        If there is only one part-ref then return a 'Part_Ref' object (or
```

Comment:
This routine only ever returns `Data_Ref` or None AFAICS.

Reply:
Yes, you're right. I was talking about the way the class behaves rather than the method.